File: daily-challenge/2025-09-23/solution.cpp
"""
Problem: 165 - Compare Version Numbers  
Description: Given two version numbers as strings, compare them and return -1 if version1 < version2, 1 if version1 > version2, or 0 if they are equal.  
Date: 2025-09-23
"""

class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i = 0, j = 0, n1 = version1.size(), n2 = version2.size();
        while (i < n1 || j < n2) {
            int num1 = 0, num2 = 0;
            while (i < n1 && version1[i] != '.') {
                num1 = num1 * 10 + (version1[i++] - '0');
            }
            while (j < n2 && version2[j] != '.') {
                num2 = num2 * 10 + (version2[j++] - '0');
            }
            if (num1 < num2) return -1;
            if (num1 > num2) return 1;
            i++; j++;
        }
        return 0;
    }
};

File: daily-challenge/2025-09-23/solution.py
"""
Problem: 165 - Compare Version Numbers  
Description: Given two version numbers as strings, compare them and return -1 if version1 < version2, 1 if version1 > version2, or 0 if they are equal.  
Date: 2025-09-23
"""

class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = list(map(int, version1.split('.')))
        v2 = list(map(int, version2.split('.')))
        max_len = max(len(v1), len(v2))
        v1 += [0] * (max_len - len(v1))
        v2 += [0] * (max_len - len(v2))
        
        for a, b in zip(v1, v2):
            if a < b: return -1
            if a > b: return 1
        return 0
